tracing/src/main/java/com/palantir/remoting2/tracing/Tracers.java

```diff
@@ -96,11 +100,14 @@ public static Runnable wrap(Runnable delegate) {
         @Override
         public V call() throws Exception {
             Trace originalTrace = Tracer.copyTrace();
+            String originalMdcTraceIdValue = MDC.get(MDC_KEY);
             Tracer.setTrace(trace);
+            MDC.put(MDC_KEY, trace.getTraceId());
```

Comment:
@uschi2000 @markelliot this PR seems like a bug fix for the current state of affairs. Is there a reason to block it while we figure out if it is worth to change the approach?